sql/opt_range.cc

```diff
@@ -2479,6 +2479,8 @@ int SQL_SELECT::test_quick_select(THD *thd, key_map keys_to_use,
       KEY_PART_INFO *key_part_info;
       uint n_key_parts= head->actual_n_key_parts(key_info);
 
+	  if (key_info->algorithm == HA_KEY_ALG_HASH)
+		  continue;
```

Comment:
Please maintain server coding style, no tabs (use spaces) and indention rules are different! See around this file.